src/backend/crossterm.rs

```diff
@@ -127,6 +119,194 @@ where
+    /// };
+    /// let terminal = CrosstermBackend::stdout().into_terminal_with_options(options)?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    pub fn into_terminal_with_options(
+        self,
+        options: TerminalOptions,
+    ) -> io::Result<Terminal<Self>> {
+        Terminal::with_options(self, options)
+    }
+
+    /// Converts the `CrosstermBackend` into a [`Terminal`] instance with default settings.
+    ///
+    /// This enables raw mode and switches to the alternate screen. Mouse support is not enabled.
+    ///
+    /// Raw mode and alternate screen are restored when the `CrosstermBackend` is dropped.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use ratatui::backend::CrosstermBackend;
+    ///
+    /// let terminal = CrosstermBackend::stdout().into_terminal_with_defaults()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    pub fn into_terminal_with_defaults(self) -> io::Result<Terminal<Self>> {
+        let backend = self.with_raw_mode()?.with_alternate_screen()?;
+        Terminal::new(backend)
+    }
+
+    /// Enables raw mode for the terminal.
+    ///
+    /// Returns an [`io::Result`] containing self so that it can be chained with other methods.
+    ///
+    /// Raw mode is restored when the `CrosstermBackend` is dropped.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use ratatui::backend::CrosstermBackend;
+    ///
+    /// let backend = CrosstermBackend::stdout().with_raw_mode()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    pub fn with_raw_mode(mut self) -> io::Result<Self> {
+        self.restore_raw_mode_on_drop = true;
+        terminal::enable_raw_mode()?;
+        Ok(self)
+    }
+
+    /// Enables raw mode for the terminal and switches to the alternate screen.
+    ///
+    /// Returns an [`io::Result`] containing self so that it can be chained with other methods.
+    ///
+    /// Alternate screen is restored when the `CrosstermBackend` is dropped.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use ratatui::backend::CrosstermBackend;
+    ///
+    /// let backend = CrosstermBackend::stdout().with_alternate_screen()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    pub fn with_alternate_screen(mut self) -> io::Result<Self> {
+        self.restore_alternate_screen_on_drop = true;
+        execute!(self.writer, terminal::EnterAlternateScreen)?;
+        Ok(self)
+    }
+
+    /// Enables mouse support for the terminal.
+    ///
+    /// Returns an [`io::Result`] containing self so that it can be chained with other methods.
+    ///
+    /// Mouse support is restored when the `CrosstermBackend` is dropped.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use ratatui::backend::CrosstermBackend;
+    ///
+    /// let backend = CrosstermBackend::stdout().with_mouse_support()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    pub fn with_mouse_support(mut self) -> io::Result<Self> {
```

Comment:
I think it should be named closer to the name of crossterm
```suggestion
    pub fn with_mouse_event_capture(mut self) -> io::Result<Self> {
```